src/algebra/lie/basic.lean

```diff
@@ -1214,10 +1226,45 @@ variables [lie_ring_module L M] [lie_module R L M]
 
 namespace lie_algebra
 
+variables (I : lie_ideal R L)
+
+/-- A generalisation of the derived series of a Lie algebra, whose zeroth term is a specified ideal.
+
+It can be more convenient to work with this generalisation when considering the derived series of
+an ideal since it provides a type-theoretic expression of the fact that the terms of the ideal's
+derived series are also ideals of the enclosing algebra.
+
+See also `lie_ideal.derived_series_eq_derived_series_of_ideal_comap` and
+`lie_ideal.derived_series_eq_derived_series_of_ideal_map` below. -/
+def derived_series_of_ideal : ℕ → lie_ideal R L
+| 0       := I
+| (k + 1) := ⁅derived_series_of_ideal k, derived_series_of_ideal k⁆
+
+@[simp] lemma derived_series_of_ideal_zero :
+  derived_series_of_ideal R L I 0 = I := rfl
+
+@[simp] lemma derived_series_of_ideal_succ (k : ℕ) :
+  derived_series_of_ideal R L I (k + 1) =
+  ⁅derived_series_of_ideal R L I k, derived_series_of_ideal R L I k⁆ := rfl
+
 /-- The derived series of Lie ideals of a Lie algebra. -/
-def derived_series : ℕ → lie_ideal R L
-| 0       := ⊤
-| (k + 1) := ⁅derived_series k, derived_series k⁆
+abbreviation derived_series : ℕ → lie_ideal R L := derived_series_of_ideal R L ⊤
+
+@[simp] lemma derived_series_def (k : ℕ) :
+  derived_series R L k = derived_series_of_ideal R L ⊤ k := rfl
```

Comment:
Does this work?
```suggestion
@[simp] abbreviation derived_series : ℕ → lie_ideal R L := derived_series_of_ideal R L ⊤
```

Reply:
Do we actually want this to be simp by default? I'm not so sure about that.

Reply:
@eric-wieser that doesn't seem to work, though it's a nice idea.

@jcommelin yes, on reflection this should not carry `simp`. For one thing I'd like to be able to state `simp` lemmas with `derived_series` on the LHS.

Incoming commit.